torch/lib/c10d/ProcessGroupWrapper.cpp

```diff
@@ -71,81 +93,75 @@ struct CollectiveFingerPrint {
       output_tensors.emplace_back(outputs);
     }
     // Allgather tensor shapes.
-    pg->allgather(output_tensors, shape_tensors)->wait();
+    pg->allgather(output_tensors, tensors_to_verify)->wait();
```

Comment:
This allgather would fail if shapes are different across processes right? We are just using the local tensor shape above ?

Reply:
This would be the case, but the way this is called this `allgather` should always succeed. This is because we call `verify_tensors()` with a tensor that actually contains only the same of the tensor that we're ultimately trying to verify. 

Basically we have a 1D tensor of size k_i on each process. First we get the shape of k_i which is just a single int, allgather that, verify it, and then only call into the allgather on the k_i tensors if that succeeds. This ensures that we call allgather in a consistent way so it doesn't crash.